src/Private/CanonicalRequest.cs

```diff
@@ -30,6 +30,24 @@ public static class CanonicalRequest
         /// </summary>
         public static string HeaderValueSeparator { get; set; } = ", ";
 
+        // Including most headers from
+        // https://github.com/smithy-lang/smithy-typescript/blob/430021abf44f8a4d6c24de2dfa25709bf91a92c8/packages/signature-v4/src/constants.ts#L19-L35
+        private static HashSet<string> UnsignableHeaders = new(StringComparer.OrdinalIgnoreCase) {
+            "authorization",
+            "connection",
+            "expect",
+            "keep-alive",
+            "proxy-authenticate",
+            "proxy-authorization",
+            "proxy-connection",
+            "range",
+            "te",
+            "trailer",
+            "transfer-encoding",
+            "upgrade",
+            HeaderKeys.XAmznTraceIdHeader,
+        };
```

Comment:
Sources I used to come up with this list:
- primarily [@smithy/signature-v4](https://github.com/smithy-lang/smithy-typescript/blob/430021abf44f8a4d6c24de2dfa25709bf91a92c8/packages/signature-v4/src/constants.ts#L19-L35) (used by AWS SDK for JavaScript)
- [AWS SDK for Java](https://github.com/aws/aws-sdk-java-v2/blob/47fcc27deb90d4102ca5b76c35d27749b1ca0e08/core/auth/src/main/java/software/amazon/awssdk/auth/signer/internal/AbstractAws4Signer.java#L70-L71)
- [AWS SDK for .NET](https://github.com/aws/aws-sdk-net/blob/b8431ed566827a081624a558d7d76b35ecf5f04d/sdk/src/Core/Amazon.Runtime/Internal/Auth/AWS4Signer.cs#L61-L66)
- [aws4fetch](https://github.com/mhart/aws4fetch/blob/629b108747e066d1b93e8b72a4adbf2a4220db3b/src/main.js#L22-L33)

The final list here is still quite conservative - all headers listed here have been verified to always cause invalid signatures with an IAM-protected API Gateway REST API endpoint.

Some headers are unsigned by the libraries referenced above, but not included here, e.g.
- AWS SDK for .NET ignores `amz-sdk-invocation-id` and `amz-sdk-request`, but only because of their specific retry behaviour, which isn't relevant here. See https://github.com/aws/aws-sdk-net/commit/9017e427ebcd0b1aa60b3ba472ba74282a3348be
- @smithy/signature-v4 skips many browser specific headers that are likely not relevant here

Quite a few libraries also ignore the `user-agent` header in the signature. I'm not sure if they encountered similar issues to https://github.com/FantasticFiasco/aws-signature-version-4/issues/1155 or if there's another reason. In any case, the user-agent header doesn't cause signature problems in the latest version of AwsSignatureVersion4 in my testing.

Reply:
Thanks for analyzing the other libraries and creating a coherent list of headers to ignore.

I've created #1208 to analyze the `User-Agent` header, potentially opening up to ignore the header instead of implementing a workaround for it.